jsontableschema/types/base.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from functools import partial
+from abc import ABCMeta, abstractmethod
+from .. import compat
+from .. import exceptions
+from .. import constraints
+
+
+# Module API
+
+class JTSType(object):
+    """Base class for all JSON Table Schema types.
+
+    Args:
+        field (dict): field schema
+
+    Aside implementing `cast_default` subclass could add other cast methods
+    with the same signature like `cast_fmt`, `cast_current` etc
+    to add support for corresponding formats.
+
+    """
+
+    # Public
+
+    @property
+    @abstractmethod
+    def name(self):
+        """str: type name like `array`
+        """
+        pass  # pragma: no cover
+
+    @property
+    @abstractmethod
+    def null_values(self):
+        """str[]: list of strings to process as null value
+        """
+        pass  # pragma: no cover
+
+    @property
+    @abstractmethod
+    def supported_constraints(self):
+        """str[]: list of supported JTS constraints
+        """
+        pass  # pragma: no cover
+
+    def __init__(self, field=None):
+
+        # Set default field
+        self.__field = None
+        self.__field_name = None
+        self.__format = 'default'
+        self.__constraints = {}
+
+        # Set user defined field
+        if field:
+            self.__field = field
+            self.__field_name = field['name']
+            self.__format = field['format']
+            self.__constraints = field.get('constraints', {})
+
+        # Set parsed format (fmt feature)
+        self.__format_main = self.__format
+        self.__format_fmt = None
+        if self.__format.startswith('fmt'):
+            self.__format_main = 'fmt'
+            self.__format_fmt = self.__format.strip('fmt:')
+
+    def cast(self, value, skip_constraints=False):
+        """Cast value.
+
+        Args:
+            value (any): value to cast
+            skip_constraints (bool): if True it skips constraints checks
```

Comment:
It was using internally to cast things like `enum` etc (so I have saved this approach):
```
                if check_name in ['minimum', 'maximum']:
                    check_value = self.cast(check_value, skip_constraints=True)
```